Client/Frontend/Browser/TabManager.swift

```diff
@@ -447,10 +438,27 @@ class TabManager: NSObject {
 
     /// Removes all private tabs from the manager without notifying delegates.
     private func removeAllPrivateTabs() {
+        tabs.forEach { tab in
+            if tab.isPrivate {
+                removeAllBrowsingDataForTab(tab)
+            }
+        }
+
         tabs = tabs.filter { !$0.isPrivate }
         delegates.forEach { $0.get()?.tabManagerDidRemoveAllTabs(self, toast: nil) }
     }
 
+    func removeAllBrowsingDataForTab(_ tab: Tab, completionHandler: @escaping () -> Void = {}) {
```

Comment:
when restoring tabs on launch do we also need to call this method for each tab?

Reply:
You mean for non-private tabs? No. It should already be getting invoked on launch for private tabs though.